source/core/slang-char-util.h

```diff
@@ -0,0 +1,51 @@
+#ifndef SLANG_CORE_CHAR_UTIL_H
+#define SLANG_CORE_CHAR_UTIL_H
+
+#include "slang-string.h"
+
+namespace Slang {
+
+struct CharUtil
+{
+    typedef uint8_t Flags;
+    struct Flag
+    {
+        enum Enum : Flags
+        {
+            Upper           = 0x01,
+            Lower           = 0x02,
+            Digit           = 0x04,         
+            LineWhitespace  = 0x08,         ///< Whitespace that can appear in a line (ie excluding CR/LF)
+            HexDigit        = 0x10,         ///< 0-9, a-f, A-F
+        };
+    };
+
+    SLANG_FORCE_INLINE static bool isDigit(char c) { return c >= '0' && c <= '9'; }
+    SLANG_FORCE_INLINE static bool isLower(char c) { return c >= 'a' && c <= 'z'; }
+    SLANG_FORCE_INLINE static bool isUpper(char c) { return c >= 'A' && c <= 'Z'; }
+    SLANG_FORCE_INLINE static bool isLineWhitespace(char c) { return c == ' ' || c == '\t'; }
+
+        /// True if it's alpha
+    SLANG_FORCE_INLINE static bool isAlpha(char c) { return (getFlags(c) & (Flag::Upper | Flag::Lower)) != 0; }
```

Comment:
Have you benchmarked that using the flags this way is faster than just doing `isLower(c) | isUpper(c)`? My intuition would be that having to do the load to index into `g_charFlagMap` would cost more than doing both comparisons and the OR.

Reply:
I have benchmarked on x64, if single range test it's slightly faster, with two ranges it is typically (although perhaps not always) faster.